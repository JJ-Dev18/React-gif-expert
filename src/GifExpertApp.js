import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GiftGrid } from './components/GiftGrid';

export const GifExpertApp = () => {
     
   
    const [categories, setCategories] = useState(['One punch']);
   
    
    return (
        <>
              
               <h2>GifExpertApp</h2>
               <AddCategory  setCategories={setCategories} categories={categories}/>
               <hr/>
               
               <ol>
                  {
                          categories.map( category => (
                           <GiftGrid 
                           key={category}
                           category = {category} />
                      ))
                  }
               </ol>
        </>
    )
}
