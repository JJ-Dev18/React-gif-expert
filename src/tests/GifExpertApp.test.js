import '@testing-library/jest-dom'
import React  from 'react'
import { shallow } from 'enzyme'
import { GifExpertApp } from '../GifExpertApp'


 describe('Prueba en el componente principal GifExpertAPP', () => {
        test('Debe cargarse correctamente el gifExpertApp', () => {
            
            const wrapper = shallow(<GifExpertApp />)

            expect(wrapper).toMatchSnapshot();
        });

        test('debe de mostrar una lista de categorias', () => { 

            const categories = ['one punch', 'dragon ball'];

            const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)

            expect(wrapper).toMatchSnapshot();

            expect(wrapper.find('GiftGrid').length).toBe(categories.length)
            
        })
        
        
 })
 
