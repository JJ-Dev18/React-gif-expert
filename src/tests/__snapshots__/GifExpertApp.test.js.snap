// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente principal GifExpertAPP Debe cargarse correctamente el gifExpertApp 1`] = `
<Fragment>
  <h1>
    Te amo Maria Camila Novoa juntosx100pre 
  </h1>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categories={Array []}
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Prueba en el componente principal GifExpertAPP debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    Te amo Maria Camila Novoa juntosx100pre 
  </h1>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categories={
      Array [
        "one punch",
        "dragon ball",
      ]
    }
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GiftGrid
      category="one punch"
      key="one punch"
    />
    <GiftGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
</Fragment>
`;
